modules/stochastic_tools/src/reporters/AdaptiveImportanceStats.C

```diff
@@ -65,8 +69,10 @@ AdaptiveImportanceStats::execute()
     return;
   }
 
+  const bool gp_flag =
+      isParamValid("flag_sample") ? getReporterValue<std::vector<bool>>("flag_sample")[0] : false;
```

Comment:
Could you store a pointer to this and check if it exists and if yes you use the value? The reporter lookup might be expensive in `execute()`. You could create it either in the constructor or in `initialSetup`.